### PR TITLE
Flatten properties

### PR DESCRIPTION
**Pull request by Porges**

### CODE REVIEW

File: hack/generator/pkg/astmodel/property_definition.go [hunk 1]
```diff
@@ -55,11 +56,26 @@ func (property *PropertyDefinition) PropertyType() Type {
 
 // SetRequired sets if the property is required or not
 func (property *PropertyDefinition) SetRequired(required bool) *PropertyDefinition {
+	if required == property.isRequired {
+		return property
+	}
+
 	result := *property
```
Comment (theunrepentantgeek) [inline hack/generator/pkg/astmodel/property_definition.go:63]: This should use `.copy()` to safely copy the `tags` map as well.

File: hack/generator/pkg/astmodel/property_definition.go [hunk 2]
```diff
@@ -55,11 +56,26 @@ func (property *PropertyDefinition) PropertyType() Type {
...
 	return &result
 }
 
+// SetFlatten sets if the property should be flattened or not
+func (property *PropertyDefinition) SetFlatten(flatten bool) *PropertyDefinition {
+	if flatten == property.flatten {
+		return property
+	}
+
+	result := *property
```
Comment (theunrepentantgeek) [inline hack/generator/pkg/astmodel/property_definition.go:74]: Ditto should use .copy() to safely copy the tags map as well.

File: hack/generator/pkg/astmodel/property_definition.go (hunk 1 above)
Comment (Porges) [reply hack/generator/pkg/astmodel/property_definition.go:63]: do we still need to copy it even though we aren't modifying the tags?

**Comment by Porges** [comment]
Reworking into different PRs.

**Pull request closed without merging**